tooling/reproducible/macos_repro_build_compare.sh

```diff
@@ -0,0 +1,686 @@
+
+    if containsElement "$fixed_param" "${CONFIG_ARGS[@]}"; then
+      # Check if fixed_param is in CONFIG_ARGS
+      STRINGTOADD="$fixed_param$fixed_value"
+      CONFIG_ARRAY+=("$STRINGTOADD")
+    elif containsElement "$fixed_param" "${NOTUSE_ARGS[@]}"; then
+      # Check if fixed_param is in NOTUSE_ARGS
+      STRINGTOADD="$fixed_param$fixed_value"
+      IGNORED_ARRAY+=("$STRINGTOADD")
+    elif containsElement "$fixed_param" "${FINAL_ARG[@]}"; then
+      # Check if parameter should be at the end, usually just version & variant
+      STRINGTOADD="$fixed_param$fixed_value"
+      FINAL_ARRAY+=("$STRINGTOADD")
+    else
+      # Neither Config Or Ignore, so Build
+      STRINGTOADD="$fixed_param$fixed_value"
+      BUILD_ARRAY+=("$STRINGTOADD")
+    fi
+  done
+
+  # Construct Final Parameter String
+  for element in "${BUILD_ARRAY[@]}"; do
+    build_string+="$element"
+  done
+
+  for element in "${CONFIG_ARRAY[@]}"; do
+    config_string+="$element"
+  done
+
+  for element in "${FINAL_ARRAY[@]}"; do
+    final_string+="$element"
+  done
+
+  final_params="$build_string --configure-args \"$config_string\" $final_string"
+
+echo ""
+  echo "Make JDK Any Platform Argument List = "
+  echo "$final_params"
+  echo ""
+  echo "Parameters Parsed Successfully"
+}
+
+Build_JDK() {
+  echo "Building JDK..."
+
+  # Trigger Build
+  cd "$WORK_DIR"
+  echo "cd temurin-build && ./makejdk-any-platform.sh $final_params 2>&1 | tee build.$$.log" | sh
+  # Copy The Built JDK To The Working Directory
+  cp $WORK_DIR/temurin-build/workspace/target/"$target_file" $WORK_DIR/reproJDK.tar.gz
+}
+
+Compare_JDK() {
+  echo "Comparing JDKs"
+  echo ""
+  mkdir "$WORK_DIR/compare"
+  cp $WORK_DIR/src_jdk_dist.tar.gz $WORK_DIR/compare
+  cp $WORK_DIR/reproJDK.tar.gz $WORK_DIR/compare
+  
+
+  # Get The Current Versions Of The Reproducible Build Scripts
+  wget -O "$WORK_DIR/compare/repro_common.sh" "https://raw.githubusercontent.com/adoptium/temurin-build/master/tooling/reproducible/repro_common.sh"
+  wget -O "$WORK_DIR/compare/repro_compare.sh" "https://raw.githubusercontent.com/adoptium/temurin-build/master/tooling/reproducible/repro_compare.sh"
+  wget -O "$WORK_DIR/compare/repro_process.sh" "https://raw.githubusercontent.com/adoptium/temurin-build/master/tooling/reproducible/repro_process.sh"
+
+  # Set Permissions
+  chmod +x "$WORK_DIR/compare/"*sh
+  cd "$WORK_DIR/compare"
+
+  # Unzip And Rename The Source JDK
+  echo "Unzip Source"
+  tar xvfz src_jdk_dist.tar.gz
+  original_directory_name=$(find . -maxdepth 1 -type d | tail -1)
+  mv "$original_directory_name" src_jdk
+
+  #Unzip And Rename The Target JDK
+  echo "Unzip Target"
+  tar xvfz reproJDK.tar.gz
+  original_directory_name=$(find . -maxdepth 1 -type d | grep -v src_jdk | tail -1)
+  mv "$original_directory_name" tar_jdk
+
+  # These Two Files Are Generate Classes And Should Be Removed Prior To Running The Comparison
```

Comment:
The removal of the classes.jsa is done by repro_common.sh now, just need to add for Darwin here: https://github.com/adoptium/temurin-build/blob/298347ac59e6b73b15d99bce58b66a6d44cce940/tooling/reproducible/repro_common.sh#L218

Reply:
Fixed.